flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowAggregateTestPrograms.java

```diff
@@ -0,0 +1,494 @@
+                                + "FROM TABLE(%s)\n"
+                                + "GROUP BY name, window_start, window_end",
+                        windowSql);
+
+        TableTestProgram.Builder builder = TableTestProgram.of(id, description);
+        if (enableDistinctAggSplit) {
+            builder.setupConfig(
+                    OptimizerConfigOptions.TABLE_OPTIMIZER_DISTINCT_AGG_SPLIT_ENABLED, true);
+        }
+
+        return builder.setupConfig(
+                        OptimizerConfigOptions.TABLE_OPTIMIZER_AGG_PHASE_STRATEGY, aggPhaseStrategy)
+                .setupTableSource(SOURCE)
+                .setupTableSink(
+                        SinkTestStep.newBuilder("sink_t")
+                                .addSchema(
+                                        "name STRING",
+                                        "window_start TIMESTAMP(3)",
+                                        "window_end TIMESTAMP(3)",
+                                        "cnt BIGINT",
+                                        "sum_int INT",
+                                        "distinct_cnt BIGINT")
+                                .consumedBeforeRestore(beforeRows)
+                                .consumedAfterRestore(afterRows)
+                                .build())
+                .runSql(sql)
+                .build();
+    }
+
+    static final TableTestProgram TUMBLE_WINDOW_PROC_TIME =
+            TableTestProgram.of(
+                            "window-aggregate-tumble-proc-time",
+                            "validates group by using tumbling window with processing time")
+                    .setupTableSource(SOURCE)
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink_t")
+                                    .addSchema(
+                                            "name STRING",
+                                            "cnt BIGINT",
+                                            "sum_int INT",
+                                            "distinct_cnt BIGINT")
+                                    .consumedBeforeRestore(
+                                            "+I[a, 6, 18, 3]",
+                                            "+I[null, 1, 7, 0]",
+                                            "+I[b, 4, 14, 3]")
+                                    .consumedAfterRestore(
+                                            "+I[a, 1, 10, 1]", "+I[c, 1, 12, 1]", "+I[d, 2, 24, 2]")
+                                    .build())
+                    .runSql(
+                            "INSERT INTO sink_t SELECT "
+                                    + "name, "
+                                    + "COUNT(*), "
+                                    + "SUM(a_int), "
+                                    + "COUNT(DISTINCT `comment`) "
+                                    + "FROM source_t "
+                                    + "GROUP BY name, TUMBLE(proctime, INTERVAL '5' SECOND)")
```

Comment:
All tests in this class are about window agg, but this test is related about group window agg, right?

Reply:
Yes!  Good catch.